File: 151001/Novik/DC/src/Discussion/Mapper/PostMapper.cs
using AutoMapper;
using Discussion.Models.DTOs.Requests;
using Discussion.Models.DTOs.Responses;
using Discussion.Models.Entity;

namespace Discussion.Mapper;

public class PostMapper : Profile
{

    public PostMapper()
    {
        CreateMap<Post, PostResponseTo>();
        CreateMap<PostRequestTo, Post>();
    }
    
}

File: 151001/Trubchik/dc/publisher/src/main/java/by/bsuir/ilya/dto/PostMapper.java
package by.bsuir.ilya.dto;

import by.bsuir.ilya.Entity.Post;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper
public interface PostMapper {
    PostMapper INSTANCE = Mappers.getMapper( PostMapper.class );

    Post postResponseToToPost(PostResponseTo responseTo);
    Post postRequestToToPost(PostRequestTo requestTo);

    PostRequestTo postToPostRequestTo(Post post);

    PostResponseTo postToPostResponseTo(Post post);

    List<PostResponseTo> listRequestToResponse(List<PostRequestTo> requestList);
}
